tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1037 @@
+        self, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        with patch("builtins.open", mock_open()) as mocked_open_file:
+            AGWInstallerNetworkConfigurator(
+                self.TEST_NETWORK_INTERFACES, None, None
+            )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+            mocked_open_file.assert_called_with("/etc/network/interfaces", "w")
+            mocked_open_file().write.assert_called_once_with(f"source-directory {ifaces_dir}")
+
+    def test_given_agw_installer_network_configurator_when_static_eth0_address_given_then_correct_eth0_configuration_is_retured(  # noqa: E501
+        self,
+    ):
+        test_static_ip_address = "1.2.3.4/24"
+        test_static_gw_ip_address = "4.3.2.1"
+        expected_config = [
+            "auto eth0",
+            "iface eth0 inet static",
+            "address 1.2.3.4",
+            "netmask 255.255.255.0",
+            "gateway 4.3.2.1",
+        ]
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES,
+            test_static_ip_address,
+            test_static_gw_ip_address,
+        )
+        self.assertEqual(agw_network_configurator._eth0_config, expected_config)
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR",  # noqa: E501
+        ".tox/.tmp",
+    )
+    def test_given_agw_installer_network_configurator_when_static_eth0_address_given_then_correct_message_is_logged_while_creating_interface_config_file(  # noqa: E501
+        self,
+    ):
+        test_static_ip_address = "1.2.3.4/24"
+        test_static_gw_ip_address = "4.3.2.1"
+        with self.assertLogs() as logs:
+            AGWInstallerNetworkConfigurator(
+                self.TEST_NETWORK_INTERFACES,
+                test_static_ip_address,
+                test_static_gw_ip_address,
+            )._prepare_eth0_configuration()
+        self.assertEqual(
+            logs.records[0].getMessage(),
+            "Preparing configuration for SGi interface (eth0)...",
+        )
+        self.assertEqual(
+            logs.records[1].getMessage(),
+            f"Using statically configured IP {test_static_ip_address} "
+            f"and Gateway {test_static_gw_ip_address}...",
+        )
+
+    def test_given_agw_installer_network_configurator_when_no_static_eth0_address_given_then_correct_eth0_configuration_is_retured(  # noqa: E501
+        self,
+    ):
+        expected_config = [
+            "auto eth0",
+            "iface eth0 inet dhcp",
+        ]
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES,
+            None,
+            None,
+        )
+        self.assertEqual(agw_network_configurator._eth0_config, expected_config)
```

Comment:
Another private attribute check? BTW, I assume the reason we don't want to check those is the same as for private methods, doesn't it?

Reply:
Tests refactored.